pkgs/tools/networking/gvproxy/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, fetchFromGitHub, buildGoModule }:
+
+buildGoModule rec {
+  pname = "gvproxy";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = "gvisor-tap-vsock";
+    rev = "v${version}";
+    sha256 = "sha256-cXaX7A8Wgq/PEHnFrparZjOKdENX3pWWXYfrEuFLDx0=";
+  };
+
+  vendorSha256 = null;
+
+  buildPhase = ''
+    runHook preBuild
+    make build
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    install bin/* -Dt $out/bin
+    runHook postInstall
+  '';
```

Comment:
I don't think we need to use the upstream makefile. It is just building 3 binaries which buildGoModule can do, too.

Reply:
>  It is just building 3 binaries which buildGoModule can do, too.

And `ldflags`. And conditional `CGO` +  `GOOS`.

Staying with the makefile.